### PR TITLE
Add example readme

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
Related to https://github.com/apache/arrow-rs/issues/1613

Having concise descriptions of examples will help people find them

# What changes are included in this PR?

`README.md` file in arrow / examples directory

See rendered version: https://github.com/alamb/arrow-rs/tree/alamb/examples_doc/arrow/examples

# Are there any user-facing changes?

moar docs!

### CODE REVIEW

File: arrow/examples/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Examples
+
+- [`builders](builders.rs): Using the Builder API
```
Comment (viirya) [inline arrow/examples/README.md:22]: Missing right "`"

Comment (viirya) [reply arrow/examples/README.md:22]: And, maybe `builders.rs`? To be consistent with others.

**Review by viirya — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @viirya

File: arrow/examples/README.md (hunk 1 above)
Comment (alamb) [reply arrow/examples/README.md]: in 6851adf073

**Pull request merged**